File: 5th Semester/Programming Paradigms/NP2 Exercises/ex4.py
# Escreve um programa que pergunte o salário do funcionário e calcule o valor do aumento. Para salários
# superiores a R$ 1250,00, calcule um aumento de 10%. Para os inferiores ou iguais, de 15%.

def ler_float(mensagem: str) -> float:
    while True:
        output = str(input(mensagem)).strip()
        try:
            return float(output.replace(',', '.'))
        except ValueError:
            print('Por favor insira um número real válido.')

def main():
    print("Programa calcula aumento de salários.")
    # Walrus operator! O seguinte programa funciona do python 3.8 para frente. Feito para tirar onda.
    #print("O salário com aumento ficará R$%.2f." % (sal * 1.15 if (sal := ler_float('Digite um salário: ')) <= 1250 else sal*1.1))
    print("O salário com aumento ficará R$%.2f." % (sal * 1.15 if (sal := float(input('Digite um salário: '))) <= 1250 else sal*1.1))

if __name__ == "__main__":
    main()

File: 5th Semester/Programming Paradigms/NP2 Exercises/ex4.c
//  Escreve um programa que pergunte o salário do funcionário e calcule o valor do aumento. Para salários
// superiores a R$ 1250,00, calcule um aumento de 10%. Para os inferiores ou iguais, de 15%.

#include <stdio.h>
#include <stdlib.h>
#define TAMANHO_OUTPUT 100

float lerFloat(char mensagem[]) {
    char output[TAMANHO_OUTPUT];
    float numFloat;
    while(1) {
        printf("%s", mensagem);
        fgets(output, TAMANHO_OUTPUT, stdin);
        if (!((numFloat = atof(output)) == 0 && output[0] != '0'))
            return numFloat;
        printf("Por favor insira um número real válido!\n");
    }
}

int main() {
    printf("Programa calcula aumento de salários.\n");
    float sal = lerFloat("Digite um salário: ");
    printf("O salário com aumento ficará %.2f\n", sal > 1250 ? sal*1.1 : sal*1.15);
    return 0;
}